torchdata/dataloader2/communication/protocol.py

```diff
@@ -81,18 +97,36 @@ def response_terminate(self):
         if not self.have_pending_request():
             raise Exception("Attempting to reply with pending request")
         if not isinstance(self._req_received, communication.messages.TerminateRequest):
-            raise Exception("Replaying with terminate status to other type of message")
+            raise Exception("Replaying with `terminate` status to other type of message")
```

Comment:
Not related to this PR. I think we might want to change the `Exception` to something more specific so that we can easily handle those `Exception` easily.

Reply:
Opened an issue: https://github.com/pytorch/data/issues/996